app/api/Torrent.js

```diff
@@ -48,8 +48,21 @@ export default class Torrent {
 
     this.engine.add(magnetURI, { path: cacheLocation }, torrent => {
       const server = torrent.createServer();
-      server.listen(port);
-      this.server = server;
+      const thisRef = this;
+
+      detect(this.port).then((_port) => {
```

Comment:
> How about `resolvedPort` instead of `thisRef`

`thisRef` is a reference to `this`, as the value of `this` is different when inside the .then() block. I think you want me to change the name of `_port` to `resolvedPort`, yeah, that can be done. What name do you want me to give for `thisRef` ?